module/epochs/qc_client.go

```diff
@@ -0,0 +1,169 @@
+package epochs
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/onflow/cadence"
+
+	"github.com/onflow/flow-core-contracts/lib/go/templates"
+
+	sdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/client"
+	sdkcrypto "github.com/onflow/flow-go-sdk/crypto"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+)
+
+// QCContractClient is a client to the QC Contract
+type QCContractClient struct {
+	accessAddress     string         // address of the access node
+	qcContractAddress string         // QuorumCertificate contract address
+	client            *client.Client // flow-go-sdk client to access node
+	account           *sdk.Account
+	accountKeyIndex   int
+	privateKey        sdkcrypto.PrivateKey
+	signer            sdkcrypto.Signer
+}
+
+// NewQCContractClient returns a new client to the QC contract
+func NewQCContractClient(privateKey sdkcrypto.PrivateKey, accountAddress string, accountKeyIndex int, accessAddress, qcContractAddress string) (*QCContractClient, error) {
+
+	address := sdk.HexToAddress(accountAddress)
+
+	// create a new instance of flow-go-sdk client
+	flowClient, err := client.New(accessAddress, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("could not create flow client: %w", err)
+	}
+
+	// get account for given address
+	account, err := flowClient.GetAccount(context.Background(), address)
+	if err != nil {
+		return nil, fmt.Errorf("could not get account: %w", err)
+	}
+
+	// check if account key index within range of keys
+	if len(account.Keys) <= accountKeyIndex {
+		return nil, fmt.Errorf("given account key index is bigger than the number of keys for this account")
+	}
+
+	// construct signer for signing transactions
+	accountKey := account.Keys[accountKeyIndex]
+	signer := sdkcrypto.NewInMemorySigner(privateKey, accountKey.HashAlgo)
+
+	return &QCContractClient{
+		accessAddress:     accessAddress,
+		qcContractAddress: qcContractAddress,
+		client:            flowClient,
+		account:           account,
+		accountKeyIndex:   accountKeyIndex,
+		privateKey:        privateKey,
+		signer:            signer,
```

Comment:
looks like what we need is the signer instead of the private key, we could let `NewQCContractClient` takes a signer object instead of the private key itself. 

And we will need abstraction around how the signer is created. Some references:
https://github.com/dapperlabs/flow-mgmt/blob/master/sporking/kms.go
https://github.com/dapperlabs/flow-mgmt/blob/153e0bdca152a12a1d1752b5e3a2ac8bb341f4aa/sporking/signer_service.go